docs/dsl_resource.md

```diff
@@ -98,3 +98,23 @@ end
 ```
 
 For a full example, see our [example resource](https://github.com/chef/inspec/blob/master/examples/profile/libraries/gordon_config.rb).
+
+## Lazy Loading
+
+As of InSpec v4.16, resources are lazily loaded into the `inspec`
+process upon use. This greatly speeds up the initial startup costs of
+the `inspec` process and only loads what you need to use (eg, `inspec
+--version` no longer runs for 10 seconds!).
+
+It does change the way the resource registry is handled in ways that
+might break some assumptions. Specifically `inspec.<resource>` isn't
+pre-populated with the core resources that InSpec ships with. If you
+make a local/custom resource of the same name, referring to the core
+resource via `inspec.<resource>` will not resolve to the core
+resource.
+
+If you do need to do this, it is better to make a local resource with
```

Comment:
I think this section needs an example pretty badly - someone writing a custom resource would not know what you mean, I think.

Reply:
1. I don't know how to do this, as mentioned yesterday in standup.
2. I don't know if I want to encourage this any more than we have to... I just want to provide a means to get someone out of the situation they got themselves into already (and I'm honestly not sure there are any beyond @james-stocks).